### PR TITLE
BUILD: update azure-pipelines for new workflow

### PR DESCRIPTION
**Pull request by joerg-schneider**
This PR makes some changes to adopt the new release workflow (see diagram below):

- adds a test_package_version unit test case, which
   - validates `__version__` from package to match versus `^(\d)+\.(\d)+\.(\d)+(rc\d)?$` 
   - fetches existing PyPi `gamma-pytools` releases from PyPi's RSS feed XML and checks that `__version__` is unreleased – **helps as a 'bump-reminder', too**
 - updates azure-pipelines.yaml 
   - branches `develop`, `release/*` will trigger builds on these branches
   - PRs to `develop`, `develop/*`, 'release/*' will trigger builds of these source branches of the PR
   - delete variable `master_or_release`, replaced with `source_is_release_branch` and `source_is_develop_branch`
   - changed behaviour of job run conditions:  the release & docs stages are running whenever either `source_is_release_branch` and `source_is_develop_branch` is true, however publishing steps in these stages only run for `source_is_release_branch`
   - some indentation, no content change

**New release workflow**:

(NOTE: not shown is the case when working on two future releases in parallel – here need to stage too using a develop/[version] branch)

grafik

### CODE REVIEW

**Review by j-ittner — DISMISSED** [review]
Looks great, thanks!

File: test/test/pytools/test_package_version.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+log = logging.getLogger(__name__)
+
+
+def test_package_version():
+    dev_version = pytools.__version__
+
+    log.info(f"Test package version – version set to: {dev_version}")
+    assert re.match(
+        r"^(\d)+\.(\d)+\.(\d)+(rc\d)?$", dev_version
```
Comment (rkdy) [inline test/test/pytools/test_package_version.py]: Not that we will need this often, but might want to have two-digit release candidates.

```suggestion
        r"^(\d)+\.(\d)+\.(\d)+(rc\d+)?$", dev_version
```

**Review by rkdy — CHANGES_REQUESTED** [review]
Also looks good to me, just a super-minor comment, thanks Jörg!

File: test/test/pytools/test_package_version.py (hunk 1 above)
Comment (joerg-schneider) [reply test/test/pytools/test_package_version.py]: Let's do, I considered it first, too :)

**Review by rkdy — APPROVED** [review]

**Pull request merged**